src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -336,6 +341,30 @@ public void ProcessBlock(Block block, ChainedHeader chainedHeader)
                     }
                 }
 
+                if (this.network.Consensus.MaxReorgLength > 0 && this.Wallet.MultiSigAddress.Transactions.Count > MinimumRetainedTransactions)
+                {
+                    var pastMaxReorg = new List<TransactionData>();
+                    foreach (TransactionData transactionData in this.Wallet.MultiSigAddress.Transactions)
+                    {
+                        // Only want to remove transactions that are spent, and the spend must have passed max reorg too
+                        if (transactionData.SpendingDetails != null
+                            && transactionData.SpendingDetails.BlockHeight != null
+                            && transactionData.SpendingDetails.BlockHeight < (chainedHeader.Height - this.network.Consensus.MaxReorgLength))
```

Comment:
tiniest nit, could calculate `finalisedHeight` as `chainedHeader.Height - this.network.Consensus.MaxReorgLength` rather than doing the calculation every time. wouldn't block over it